cpp/src/arrow/ipc/writer.cc

```diff
@@ -563,23 +563,48 @@ Status WriteLargeRecordBatch(const RecordBatch& batch, int64_t buffer_start_offs
                           pool, kMaxNestingDepth, true);
 }
 
-Status WriteTensor(const Tensor& tensor, io::OutputStream* dst, int32_t* metadata_length,
-                   int64_t* body_length) {
-  if (!tensor.is_contiguous()) {
-    return Status::Invalid("No support yet for writing non-contiguous tensors");
+Status WriteStridedTensorData(int dim_index, int64_t offset, int elem_size,
+                              const Tensor& tensor, io::OutputStream* dst) {
+  // TODO(pcm): optimize recursion base case by doing bulk write if possible
```

Comment:
Let me go ahead and do this. We can allocate a single scratch space buffer for use for each of the inner writes